src/synapse_tools/configure_synapse.py

```
@@ -1049,6 +1068,11 @@ def get_my_grouping(grouping_type: str) -> str:
         return fd.read().strip()
 
 
+def get_envoy_migration_config(path: str) -> EnvoyMigrationConfig:
+    with open(path) as f:
```

Please note that the file could be absent on a fresh hosts, we need to make sure that Envoy will crash-loop or something like that and not stuck until the file will be created.

---

configure_synapse runs as a cron job once per minute so it should keep trying until the file exists. Do you think it's worth handling the case where the file doesn't exist? I can add that logic in, but also don't want to mask real errors (i.e. where the file is missing for a long time incorrectly).

---

I think crashing is fine, it just need to be able to properly recover after that.